bedboss/bedstat/bedstat.py

```diff
@@ -0,0 +1,229 @@
+        return str(round(size_in_bytes / 1024, 2)) + "KB"
+    elif size_in_bytes in range(1024 * 1024, 1024 * 1024 * 1024):
+        return str(round(size_in_bytes / (1024 * 1024))) + "MB"
+    elif size_in_bytes >= 1024 * 1024 * 1024:
+        return str(round(size_in_bytes / (1024 * 1024 * 1024))) + "GB"
+
+
+def bedstat(
+    bedfile: str,
+    bedbase_config: str,
+    genome: str,
+    outfolder: str,
+    ensdb: str = None,
+    open_signal_matrix: str = None,
+    bigbed: str = None,
+    sample_yaml: str = None,
+    just_db_commit: bool = False,
+    no_db_commit: bool = False,
+    force_overwrite: bool = False,
+    pm: pypiper.PipelineManager = None,
+    **kwargs,
+) -> NoReturn:
+    """
+    Run bedstat pipeline. Can be used without running from command line
+    :param str bedfile: a full path to bed file to process
+    :param str bigbed: a full path to bigbed
+    :param str bedbase_config: a path to the bedbase configuration file
+    :param str open_signal_matrix: a full path to the openSignalMatrix
+        required for the tissue specificity plots
+    :param str outfolder: folder for storing results of the pipeline
+    :param str genome: genome assembly of the sample
+    :param str ensdb: a full path to the ensdb gtf file required for genomes
+        not in GDdata
+    :param str sample_yaml: a yaml config file with sample attributes to pass
+        on more metadata
+        into the database
+    :param bool just_db_commit: whether just to commit the JSON to the database
+    :param bool no_db_commit: whether the JSON commit to the database should be
+        skipped
+    :param bool force_overwrite: whether to overwrite the existing record
+    :param pm: pypiper object
+    """
+    outfolder_stats = os.path.join(outfolder, "output", "bedstat_output")
+    bbc = bbconf.BedBaseConf(config_path=bedbase_config, database_only=True)
+    # outfolder_stats = bbc.get_bedstat_output_path()
+
+    bed_digest = md5(open(bedfile, "rb").read()).hexdigest()
+    bedfile_name = os.path.split(bedfile)[1]
+
+    fileid = os.path.splitext(os.path.splitext(bedfile_name)[0])[0]
+    outfolder = os.path.abspath(os.path.join(outfolder_stats, bed_digest))
+    json_file_path = os.path.abspath(os.path.join(outfolder, fileid + ".json"))
+    json_plots_file_path = os.path.abspath(
+        os.path.join(outfolder, fileid + "_plots.json")
+    )
+    bed_relpath = os.path.relpath(
+        bedfile,
+        os.path.abspath(os.path.join(outfolder_stats, os.pardir, os.pardir)),
+    )
+    bigbed_relpath = os.path.relpath(
+        os.path.join(bigbed, fileid + ".bigBed"),
+        os.path.abspath(os.path.join(outfolder_stats, os.pardir, os.pardir)),
+    )
+    if not just_db_commit:
+        if force_overwrite:
+            new_start = True
+
+        if not pm:
+            pm = pypiper.PipelineManager(
+                name="bedstat-pipeline",
+                outfolder=outfolder,
+            )
+
+        # run Rscript
```

Comment:
This is self-explanatory -- can you add some more detail to this comment? What is this doing?